src/utils/confighandler.cpp

```diff
@@ -4,380 +4,175 @@
+ *
+ *   KEY TYPE DEFAULT_VALUE
+ *
+ * NOTE: Please keep it well structured
+ */
+static QMap<class QString, QSharedPointer<ValueHandler>>
+  recognizedGeneralOptions = {
+      // clang-format off
+    OPTION("showHelp"                    ,Bool               ( true          )),
+    OPTION("showSidePanelButton"         ,Bool               ( true          )),
+    OPTION("showDesktopNotification"     ,Bool               ( true          )),
+    OPTION("disabledTrayIcon"            ,Bool               ( false         )),
+    OPTION("historyConfirmationToDelete" ,Bool               ( true          )),
+    OPTION("checkForUpdates"             ,Bool               ( true          )),
+#if defined(Q_OS_MACOS)
+    CUSTOM("startupLaunch"               ,Bool               ( false         )),
+#else
+    OPTION("startupLaunch"               ,Bool               ( true          )),
+#endif
+    OPTION("showStartupLaunchMessage"    ,Bool               ( true          )),
+    OPTION("copyAndCloseAfterUpload"     ,Bool               ( true          )),
+    OPTION("copyPathAfterSave"           ,Bool               ( false         )),
+#if !defined(Q_OS_MACOS) // TODO is this the right way?
+    OPTION("useJpgForClipboard"          ,Bool               ( false         )),
+#endif
+    // TODO obsolete?
+    OPTION("saveAfterCopy"               ,Bool               ( false         )),
+    OPTION("savePath"                    ,ExistingDir        (               )),
+    OPTION("savePathFixed"               ,Bool               ( false         )),
+    OPTION("uploadHistoryMax"            ,LowerBoundedInt(1  , 25            )),
+    OPTION("undoLimit"                   ,BoundedInt(1, 999  , 100           )),
+    // Interface tab
+    OPTION("uiColor"                     ,Color              ( {116, 0, 150} )),
+    OPTION("contrastUiColor"             ,Color              ( {39, 0, 50}   )),
+    OPTION("contrastOpacity"             ,BoundedInt(0, 255  , 190           )),
+    OPTION("buttons"                     ,ButtonList         ( {}            )),
+    // Filename Editor tab
+    OPTION("filenamePattern"             ,FilenamePattern    ( {}            )),
+    // Others
+    // TODO obsolete?
+    OPTION("saveAfterCopyPath"           ,ExistingDir        (               )),
+    OPTION("drawThickness"               ,LowerBoundedInt(1  , 3             )),
+    OPTION("drawColor"                   ,Color              ( Qt::red       )),
+    OPTION("userColors"                  ,UserColors         (               )),
+    OPTION("drawFontSize"                ,LowerBoundedInt(1  , 8             )),
+    OPTION("ignoreUpdateToVersion"       ,String             ( ""            )),
+    OPTION("keepOpenAppLauncher"         ,Bool               ( false         )),
+    OPTION("fontFamily"                  ,String             ( ""            )),
+    OPTION("setSaveAsFileExtension"      ,String             ( ""            )),
+      // clang-format on
+  };
```

Comment:
Please review this even if you are not a programmer to see if I have messed up some default values.

Reply:
Great idea!